osquery/core/database/rocksdb_migration.h

```diff
@@ -0,0 +1,33 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <osquery/expected.h>
+#include <rocksdb/db.h>
+
+#include <unordered_map>
+
+namespace osquery {
+
+enum class RocksdbMigrationError {
+  InvalidArgument = 1,
+  FailToOpen = 2,
+  FailToGetVersion = 3,
+  NoMigrationFromCurrentVersion = 5,
+  MigrationLogicError = 6,
+  FailToOpenSrcDatabase = 7,
+  FailMoveDatabase = 8,
+};
+
+static ExpectedSuccess<RocksdbMigrationError> migrateDatabase(
```

Comment:
The function name does'n consist of "rocksdb" or something like this. Is it suppose to work with other types of DB?

If it is "rockdb" specific I'd like to suggest create a new namespace for the function declaration and error type.
```c++
namespace rocksdb_migration {
enum class Error {...};
ExpectedSuccess<Error> migrate(...);
}
```
What do you think?